components/brave_wallet_ui/components/desktop/popup-modals/confirm-password-modal/confirm-password-modal.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+// utils
+import { getLocale } from '../../../../../common/locale'
+
+// hooks
+import { usePasswordAttempts } from '../../../../common/hooks/use-password-attempts'
+
+// components
+import { PopupModal } from '../..'
+import { NavButton } from '../../../extension'
+import { PasswordInput } from '../../../shared'
+
+// style
+import {
+  Row,
+  VerticalSpace
+} from '../../../shared/style'
+import { Title } from '../style'
+import {
+  modalWidth,
+  StyledWrapper
+} from './confirm-password-modal.style'
+
+interface Props {
+  title: string
+  onClose: () => void
+  onSuccess: () => void
+}
+
+export const ConfirmPasswordModal = ({
+  onClose,
+  onSuccess,
+  title
+}: Props) => {
+  // state
+  const [password, setPassword] = React.useState<string>('')
+  const [isCorrectPassword, setIsCorrectPassword] = React.useState<boolean>(true)
+
+  // custom hooks
+  const { attemptPasswordEntry } = usePasswordAttempts({ maxAttempts: 3 })
```

Comment:
how is maxAttempts enforced? can the user just reload and get 3 more attempts?

Reply:
If there was someway to manipulate the react state locally then I believe the attacker could get unlimited retries. I'm not aware of how this would be done and did test it a little bit during the onboarding V2 work to try and bypass the ToCs, but couldn't find anything that allowed this to occur. Maybe they could execute [this function](https://github.com/brave/brave-core/blob/7fd9cef8608edeab33ecef9470b814d467c40800/components/brave_wallet_ui/common/hooks/use-password-attempts.ts#L61) locally and do a bypass, but I'm not aware of any way to do this.

If we enforce this at the API layer though then this definitely shouldn't be a concern unless someone is manually overriding memory in which case they could probably just extract the private keys anyways since the keyring is unlocked at the point when this code is executed.